src/compiler/checker.ts

```diff
@@ -13595,7 +13595,30 @@ namespace ts {
         }
 
         function getRestTypeElementFlags(node: RestTypeNode | NamedTupleMember) {
-            return getArrayElementTypeNode(node.type) ? ElementFlags.Rest : ElementFlags.Variadic;

```

Comment:
I think editing `getArrayElementTypeNode` directly rather than `getRestTypeElementFlags` would be a better path, and would update more locations to implicitly allow both array type references any array type nodes. Plus, if you _don't_ change it there, I don't think it's properly handled in single-element-tuple rest positions.